File: obsidian-ai-assistant/api/utils/response.js
/**
 * Response Utilities
 * 
 * Utilities for formatting API responses
 */

/**
 * Format a successful response
 * 
 * @param {any} data - Response data
 * @param {number} statusCode - HTTP status code
 * @returns {Object} - Formatted response
 */
function formatResponse(data, statusCode = 200) {
  return {
    statusCode,
    body: JSON.stringify(data),
    headers: {
      'Content-Type': 'application/json',
      'Access-Control-Allow-Origin': '*',
      'Access-Control-Allow-Credentials': true
    }
  };
}

/**
 * Format an error response
 * 
 * @param {Error} error - Error object
 * @param {number} statusCode - HTTP status code
 * @returns {Object} - Formatted error response
 */
function formatError(error, statusCode = 500) {
  const errorResponse = {
    code: error.code || 'INTERNAL_SERVER_ERROR',
    message: error.message || 'An unexpected error occurred'
  };
  
  if (error.details && process.env.NODE_ENV !== 'production') {
    errorResponse.details = error.details;
  }
  
  return {
    statusCode,
    body: JSON.stringify(errorResponse),
    headers: {
      'Content-Type': 'application/json',
      'Access-Control-Allow-Origin': '*',
      'Access-Control-Allow-Credentials': true
    }
  };
}

/**
 * Format a validation error response
 * 
 * @param {string} message - Error message
 * @param {Object} details - Validation error details
 * @returns {Object} - Formatted validation error response
 */
function formatValidationError(message, details) {
  return formatError(
    {
      code: 'VALIDATION_ERROR',
      message: message || 'Validation failed',
      details
    },
    400
  );
}

/**
 * Format a not found error response
 * 
 * @param {string} message - Error message
 * @returns {Object} - Formatted not found error response
 */
function formatNotFoundError(message) {
  return formatError(
    {
      code: 'NOT_FOUND',
      message: message || 'Resource not found'
    },
    404
  );
}

module.exports = {
  formatResponse,
  formatError,
  formatValidationError,
  formatNotFoundError
}; 

File: obsidian-ai-assistant/api/response.py
import json
from typing import Any, Dict, Optional, Union

def _create_response(status_code: int, body: Dict[str, Any], headers: Optional[Dict[str, str]] = None) -> Dict[str, Any]:
    """Create a standardized API Gateway response.
    
    Args:
        status_code (int): HTTP status code
        body (dict): Response body
        headers (dict, optional): Additional headers to include
        
    Returns:
        dict: Formatted API Gateway response
    """
    default_headers = {
        "Content-Type": "application/json",
        "Access-Control-Allow-Origin": "*",  # Enable CORS
        "Access-Control-Allow-Headers": "Content-Type,X-Api-Key,Authorization",
        "Access-Control-Allow-Methods": "OPTIONS,POST,GET,PUT,DELETE"
    }
    
    if headers:
        default_headers.update(headers)
    
    return {
        "statusCode": status_code,
        "headers": default_headers,
        "body": json.dumps(body)
    }

def success_response(status_code: int = 200, data: Optional[Any] = None, message: Optional[str] = None) -> Dict[str, Any]:
    """Format a successful API response.
    
    Args:
        status_code (int): HTTP status code (default: 200)
        data (Any, optional): Response data
        message (str, optional): Success message
        
    Returns:
        dict: Formatted success response
    """
    body = {
        "success": True
    }
    
    if data is not None:
        body["data"] = data
    if message:
        body["message"] = message
        
    return _create_response(status_code, body)

def error_response(
    status_code: int,
    message: str,
    error_code: Optional[str] = None,
    details: Optional[Union[str, Dict[str, Any]]] = None
) -> Dict[str, Any]:
    """Format an error API response.
    
    Args:
        status_code (int): HTTP status code
        message (str): Error message
        error_code (str, optional): Error code for client reference
        details (Union[str, dict], optional): Additional error details
        
    Returns:
        dict: Formatted error response
    """
    body = {
        "success": False,
        "error": {
            "message": message
        }
    }
    
    if error_code:
        body["error"]["code"] = error_code
    if details:
        body["error"]["details"] = details
        
    return _create_response(status_code, body)

def validation_error(message: str, details: Optional[Dict[str, Any]] = None) -> Dict[str, Any]:
    """Format a validation error response.
    
    Args:
        message (str): Validation error message
        details (dict, optional): Validation error details
        
    Returns:
        dict: Formatted validation error response
    """
    return error_response(
        400,
        message,
        error_code="VALIDATION_ERROR",
        details=details
    ) 